catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,604 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.transformer.mgmp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.catalog.core.api.types.Isr;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.XstreamPathValueTracker;
+import org.codice.ddf.spatial.ogc.csw.catalog.transformer.GmdTransformer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.thoughtworks.xstream.io.path.Path;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Location;
+import ddf.catalog.data.types.Media;
+import ddf.catalog.transform.CatalogTransformerException;
+
+public class MgmpTransformer extends GmdTransformer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MgmpTransformer.class);
+
+    private static final String GCO_PREFIX = "gco:";
+
+    private static final String SRV_PREFIX = "srv:";
+
+    private static final String GML_PREFIX = "gml:";
+
+    private static final String MGMP_PREFIX = "mgmp:";
+
+    private MetacardType mgmpMetacardType;
+
+    public MgmpTransformer(MetacardType metacardType) {
+        super(metacardType);
+        mgmpMetacardType = metacardType;
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream)
+            throws IOException, CatalogTransformerException {
+        return super.transform(inputStream);
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream, String id)
+            throws IOException, CatalogTransformerException {
+        return super.transform(inputStream, id);
+    }
+
+    @Override
+    public LinkedHashSet<Path> buildPaths() {
+        LinkedHashSet<Path> paths = new LinkedHashSet<>();
+
+        Arrays.asList(MgmpConstants.RESOURCE_ORIGINATOR_SECURITY_PATH,
```

Comment:
Change this to `Stream.of(MgmpConstants.RESOURCE_ORIGINATOR_SECURITY_PATH,...` to avoid the unnecessary creation of the temporary list.